lib/kvdb/ikvdb.c

```diff
@@ -2863,8 +2862,10 @@ ikvdb_kvs_cursor_update_view(struct hse_kvs_cursor *cur, unsigned int flags)
     if (ev(cur->kc_err))
         return cur->kc_err;
 
-    if (ev(cur->kc_ctxn))
-        return merr(EINVAL);
+    /* This is a no-op for a transaction cursor.
+     */
+    if (cur->kc_bind)
```

Comment:
Should we return EOPNOTSUP here instead of 0?

Reply:
My only reason is that a user could think a successful call would actually update the view

Reply:
The original EINVAL was strange, because we document that an update_view call on a txn cursor is supposed to be a no-op. Since a txn's view is fixed, I guess it does make sense that updating a cursor is a no-op.

Reply:
Hmmm ok, as long as it is documented. Feels like a footgun if someone doesn't read the docs properly however.